settings/settings.php

```diff
@@ -96,6 +88,7 @@ public function register_settings_modules() {
 		 * @param array $modules the list of module classes
 		 */
 		$modules = apply_filters( 'pll_settings_modules', $modules );
+		ksort( $modules );
```

Comment:
This is a good thing that we may be able to register settings independently for each module.

But I would advocate to use an `$priority` property to sort them in the settings view, because setting manually the numeric indexes in an array could result in entries overriding others. I have coded such a sorting function for [PLL_Bulk_Translate::display_form()](https://github.com/polylang/polylang-pro/blob/c48d41da19babe9f07dd13db5bb984e0bc440be5/modules/bulk-translate/bulk-translate.php#L295-L300), that could be a good thing that we use the same logic and names (you can change them in the `bulk-translate` module if you find something more fitting).

Reply:
You are right. That's a big issue of what I have written. Especially `pll_settings_modules` is a filter that can be used by 2 plugin developpers who don't know about the other.

We could use, for each module:
```php
array(
'class' => 'class_name',
'priority' => integer,
)
```
But we have to find some way to be backward compatible. That could be not too difficult if a developper just added a class. But if a 3rd party developper attempted to replace a class as I did here: https://github.com/polylang/polylang-pro/blob/0fb329d7fffe3b3b595710c8487379be3ccc0a4c/modules/media/load.php#L10-L11, I fear that there is no proper solution.

Reply:
On GitHub, I found only this repo using the filter: https://github.com/washingtonstateuniversity/WSUWP-Extended-Polylang/blob/4a5bcf23f395cb50b9cc2b8b1bea2c483a055352/includes/wsuwp-extended-polylang.php#L32-L46. No plugins or themes are using the filter in wp.org directory.

Reply:
> But we have to find some way to be backward compatible. That could be not too difficult if a developper just added a class. But if a 3rd party developper attempted to replace a class as I did here: https://github.com/polylang/polylang-pro/blob/0fb329d7fffe3b3b595710c8487379be3ccc0a4c/modules/media/load.php#L10-L11, I fear that there is no proper solution.

If a 3rd party developer as replaced a Polylang Pro settings' module by accessing it through its numeric id in the `$modules` array, this PR already breaks i compatibility because the numeric ids have changed.

One way to keep compatibility could be to set the priority as a property of the `PLL_Settings_Module` subclasses, with a default set in the parent class. You can then keep the `$modules` array in the exact same order, and sort the modules in the [PLL_Settings_Table::prepare_items()](https://github.com/polylang/polylang/blob/9405a129d61c7ddb719ffc2b18562e51992e8e89/settings/table-settings.php#L169-L172) method, just before they are displayed.

Reply:
> On GitHub, I found only one public repo using the filter: https://github.com/washingtonstateuniversity/WSUWP-Extended-Polylang/blob/4a5bcf23f395cb50b9cc2b8b1bea2c483a055352/includes/wsuwp-extended-polylang.php#L32-L46.

It seems that the goal of this code is only to not include the Pro version settings, which is what these modification will do. We could tell the plugin authors that this code is not needed anymore once we've merged.

Reply:
> But we have to find some way to be backward compatible. That could be not too difficult if a developper just added a class. But if a 3rd party developper attempted to replace a class as I did here: https://github.com/polylang/polylang-pro/blob/0fb329d7fffe3b3b595710c8487379be3ccc0a4c/modules/media/load.php#L10-L11, I fear that there is no proper solution.

I think we will have to define `$priority` property in Polylang Pro `PLL_Settings_Advanced_Media` class.
The `pll_settings_modules` allows to replace the Polylang  `PLL_Settings_Media` class by Polylang Pro `PLL_Settings_Advanced_Media` class but not with the same order to display it in the settings list table.